lib/command.c

```diff
@@ -1962,7 +1962,8 @@ DEFUN (config_hostname,
 	struct cmd_token *word = argv[1];
 
 	if (!isalnum((int)word->arg[0])) {
-		vty_out(vty, "Please specify string starting with alphabet\n");
+		vty_out(vty,
+		    "Please specify string starting with alphabet or number\n");
```

Comment:
if the length of a hostname is limited to < 255 characters then let's actually look for this here and prevent someone from doing this.